### PR TITLE
gh-122573: Require Python 3.10 or newer for Windows builds

### PR DESCRIPTION
**Pull request by msprotz**
* Issue: gh-122573

### CODE REVIEW

File: PCbuild/find_python.bat [hunk 1]
```diff
@@ -42,12 +42,12 @@
 @if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" ("%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul) && (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") && (set _Py_Python_Source=found in externals directory) && goto :found || rmdir /Q /S "%_Py_EXTERNALS_DIR%\pythonx86"
```
Comment (Eclips4) [inline PCbuild/find_python.bat]: Shouldn't we also update this from 3.8 to 3.10?

Comment (msprotz) [reply PCbuild/find_python.bat]: Good catch! thank you, this is now fixed.

File: Misc/NEWS.d/next/Build/2024-08-01-10-55-15.gh-issue-122573.4-UCFY.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The Windows build of CPython now requires 3.10 as a minimum version, because
+of the use of match case statements within the code of Python itself.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Build/2024-08-01-10-55-15.gh-issue-122573.4-UCFY.rst]:
```suggestion
The Windows build of CPython now requires 3.10 or newer.
```

Moreover, I suggest moving this to the _Windows_ NEWS category.

**Review by gpshead — APPROVED** [review]

File: Misc/NEWS.d/next/Build/2024-08-01-10-55-15.gh-issue-122573.4-UCFY.rst (hunk 2 above)
Comment (zooba) [reply Misc/NEWS.d/next/Build/2024-08-01-10-55-15.gh-issue-122573.4-UCFY.rst]: It's merged now, but I'd have suggested keeping it in Build as it doesn't have any impact on users of a pre-built Python (who, on Windows, far exceed the number of people who build).

**Pull request merged**